synapse/http/federation/matrix_federation_agent.py

```diff
@@ -323,12 +323,19 @@ async def _resolve_server(self) -> List[Server]:
         if port or _is_ip_literal(host):
             return [Server(host, port or 8448)]
 
+        logger.debug("Looking up SRV record for %s", host.decode())
```

Comment:
this will throw an exception if `host` contains invalid utf-8. 

I'm not sure if that is a real concern here, but as a rule best to user `decode(errors='replace')` or somesuch for logging.

Reply:
I don't think it is a real concern here since `host` has to be ascii? (Probably?) But yeah using something like that sounds better. :)

Reply:
I also don't think we can get to this point with an undecodable host... but anyway, still makes a lot of sense to do that. Changed.